pkg/csi/service/common/commonco/k8sorchestrator/topology.go

```diff
@@ -738,63 +740,43 @@ func (volTopology *nodeVolumeTopology) GetNodeTopologyLabels(ctx context.Context
 	log := logger.GetLogger(ctx)
 
 	var err error
-	if volTopology.isCSINodeIdFeatureEnabled && volTopology.clusterFlavor == cnstypes.CnsClusterFlavorVanilla {
-		csiNodeTopology := &csinodetopologyv1alpha1.CSINodeTopology{}
-		csiNodeTopologyKey := types.NamespacedName{
-			Name: nodeInfo.NodeName,
-		}
-
-		// Get CsiNodeTopology instance
-		err = volTopology.csiNodeTopologyK8sClient.Get(ctx, csiNodeTopologyKey, csiNodeTopology)
-		if err != nil {
-			if apierrors.IsNotFound(err) {
-				err = createCSINodeTopologyInstance(ctx, volTopology, nodeInfo)
-				if err != nil {
-					return nil, logger.LogNewErrorCodef(log, codes.Internal, err.Error())
-				}
-			} else {
-				msg := fmt.Sprintf("failed to get CsiNodeTopology for the node: %q. Error: %+v", nodeInfo.NodeName, err)
-				return nil, logger.LogNewErrorCodef(log, codes.Internal, msg)
-			}
-		} else {
-			// If CSINodeTopology instance already exists, check if the NodeUUID
-			// parameter in Spec is populated. If not, patch the instance.
-			if csiNodeTopology.Spec.NodeUUID == "" ||
-				csiNodeTopology.Spec.NodeUUID != nodeInfo.NodeID {
-				if csiNodeTopology.Spec.NodeUUID == "" {
-					log.Infof("CSINodeTopology instance: %q with empty nodeUUID found. "+
-						"Patching the instance with nodeUUID", nodeInfo.NodeName)
-				} else {
-					log.Infof("CSINodeTopology instance: %q with different "+
-						"nodeUUID: %s found. Patching the instance with nodeUUID: %s",
-						nodeInfo.NodeName, csiNodeTopology.Spec.NodeUUID, nodeInfo.NodeID)
-				}
-				patch := []byte(fmt.Sprintf(`{"spec":{"nodeID":"%s","nodeuuid":"%s"}}`, nodeInfo.NodeName, nodeInfo.NodeID))
-				// Patch the CSINodeTopology instance with nodeUUID
-				err = volTopology.csiNodeTopologyK8sClient.Patch(ctx,
-					&csinodetopologyv1alpha1.CSINodeTopology{
-						ObjectMeta: metav1.ObjectMeta{
-							Name: nodeInfo.NodeName,
-						},
-					},
-					client.RawPatch(types.MergePatchType, patch))
-				if err != nil {
-					msg := fmt.Sprintf("Fail to patch CsiNodeTopology for the node: %q "+
-						"with nodeUUID: %s. Error: %+v",
-						nodeInfo.NodeName, nodeInfo.NodeID, err)
-					return nil, logger.LogNewErrorCodef(log, codes.Internal, msg)
-				}
-				log.Infof("Successfully patched CSINodeTopology instance: %q with Uuid: %q",
-					nodeInfo.NodeName, nodeInfo.NodeID)
-			}
+	csiNodeTopology := &csinodetopologyv1alpha1.CSINodeTopology{}
+	csiNodeTopologyKey := types.NamespacedName{
+		Name: nodeInfo.NodeName,
+	}
+	err = volTopology.csiNodeTopologyK8sClient.Get(ctx, csiNodeTopologyKey, csiNodeTopology)
+	topologyFound := true
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			msg := fmt.Sprintf("failed to get CsiNodeTopology for the node: %q. Error: %+v", nodeInfo.NodeName, err)
+			return nil, logger.LogNewErrorCodef(log, codes.Internal, msg)
 		}
-	} else {
+		topologyFound = false
```

Comment:
Instead of using `topologyFound` variable, can we use an `else` block for lines 763:779. The `updateNodeIDForTopology` needs to run whenever `volTopology.csiNodeTopologyK8sClient.Get` is successful, adding additional variable `topologyFound`  seems redundant.

Reply:
hmm, is this a style issue or a logic error that I am missing?

Reply:
IMO the variable naming `topologyFound` is incorrect here. We create CSINodeTopology instances for non-topology environments as well and mark the topology labels as empty. That is why the `else` block approach made more sense to me

Reply:
I have renamed `topologyFound` to `csiNodeTopologyFound` to be clearer.